src/Entity/OgRole.php

```diff
@@ -177,8 +177,39 @@ public function setRoleType($role_type) {
   /**
    * {@inheritdoc}
    */
+  public function getName() {
+    // If the name is not set yet, try to derive it from the ID.
+    if (empty($this->name) && !empty($this->id())) {
+      list(, , $name) = explode('-', $this->id());
+      $this->setName($name);
+    }
+    return $this->name;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setName($name) {
+    $this->name = $name;
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
   public function save() {
-    if ($this->isNew()) {
+    // The ID of a new OgRole has to consist of the entity type ID, bundle ID
```

Comment:
> I have an admin type role, and this is available for on on all groups I create.

Not sure I understand, but to clarify the logic in OG7 that should be ported to OG8:

OgRoles could be either perBundle or perGroupId.

* per bundle means the group's roles are "read-only" for the group admins, and are controlled by the site admin.
* per group Id means the group's roles are being handled by the group admins. They can add/ change existing roles - apart of course of the non-member and member.